components/scream/src/physics/p3/atmosphere_microphysics.cpp

```diff
@@ -288,7 +297,8 @@ void P3Microphysics::initialize_impl (const RunType /* run_type */)
   p3_postproc.set_variables(m_num_cols,nk_pack,prog_state.th,pmid,T_atm,t_prev,
       prog_state.qv, prog_state.qc, prog_state.nc, prog_state.qr,prog_state.nr,
       prog_state.qi, prog_state.qm, prog_state.ni,prog_state.bm,qv_prev,
-      diag_outputs.diag_eff_radius_qc,diag_outputs.diag_eff_radius_qi);
+      diag_outputs.diag_eff_radius_qc,diag_outputs.diag_eff_radius_qi,
+      diag_outputs.precip_liq_surf,diag_outputs.precip_ice_surf,precip_liq_surf,precip_ice_surf);
```

Comment:
Apologies if I'm just not reading the code deeply enough, but my understanding was that p3_main would only know about precip_flux variables and the AD would only know about precip_mass variables and this interface layer is the only place that would know about and translate between these versions. I don't see how this layer is doing that. Maybe renaming the variables _mass and _flux like I suggested above would help!

Reply:
The naming convention is probably to blame for the confusion.  What I've done here is create a local precip_liq/ice_surf (the flux leaving the surface) variable that will be used inside p3_main which is stored in the memory buffer of the interface.  Then I pass to the post-process structure the local and field managed versions so the field managed version can accumulate the precip from this substep.  The `diag_outputs` versions are the ones used by p3_main.  The other two are pointing to the field managed.

Reply:
Ok, that was my expectations. Using _flux and _mass would help clarify.